MVCapplication/src/main/java/com/softserve/if072/mvcapp/controller/ProductPageController.java

```diff
@@ -91,40 +91,141 @@ public String addProduct(ModelMap model){
-        //Category category = restTemplate.getForObject(categoryByIdUri, Category.class, param);
+        Category category = restTemplate.getForObject(categoryByIdUri, Category.class, param);
+
+        param.clear();
+        param.put("unitId", product.getUnit().getId());
+        Unit unit = restTemplate.getForObject(UnitByIdUri, Unit.class, param);
 
         product.setUser(user);
         product.setEnabled(true);
-        product.setCategory(null);
+        product.setCategory(category);
         product.setImage(null);
-        product.setUnit(unitResult[product.getUnit().getId()]);
+        product.setUnit(unit);
 
         restTemplate.postForObject(uri, product, Product.class);
 
         return "redirect:/product/";
     }
 
+    @RequestMapping(value = "/editProduct", method = RequestMethod.GET)
+    public String editProduct(@RequestParam int id, ModelMap model) {
+
+        int userId = 1;
+
+        final String uri = productUrl + "/{id}";
+        final String unitUri = unitUrl + "/";
+        final String categoryUri = categoryUrl + "{userId}";
+
+        Map<String, Integer> param = new HashMap<>();
+        param.put("id", id);
+        RestTemplate restTemplate = new RestTemplate();
+        Product product = restTemplate.getForObject(uri, Product.class, param);
+
+        Unit[] unitResult = restTemplate.getForObject(unitUri, Unit[].class);
+        List<Unit> units = Arrays.asList(unitResult);
+
+        param.clear();
+        param.put("userId", userId);
+        Category[] categoryResult = restTemplate.getForObject(categoryUri, Category[].class, param);
+        List<Category> categories = Arrays.asList(categoryResult);
+
+        if(product.getImage() == null) {
+            imageIdForEditPage = 0;
+        } else {
+            imageIdForEditPage = product.getImage().getId();
+        }
+
+        System.out.println("Method GET" + product);
+
+        model.addAttribute("units", units);
+        model.addAttribute("categories", categories);
+        model.addAttribute("product", product);
+
+        return "editProduct";
+    }
+
+    @RequestMapping(value = "/editProduct", method = RequestMethod.POST)
+    public String editProduct(@ModelAttribute("product") Product product) {
+
+        final String categoryByIdUri = categoryUrl + "/id/{categoryId}";
+        final String UnitByIdUri = unitUrl + "/{unitId}";
+        final String uri = productUrl +"/";
+
+        User user = new User();
+        user.setId(1);
+
+        RestTemplate restTemplate = new RestTemplate();
+
+        Map<String, Integer> param = new HashMap<String, Integer>();
+        param.put("categoryId", product.getCategory().getId());
+        Category category = restTemplate.getForObject(categoryByIdUri, Category.class, param);
+
+        param.clear();
+        param.put("unitId", product.getUnit().getId());
+        Unit unit = restTemplate.getForObject(UnitByIdUri, Unit.class, param);
+
+        if(imageIdForEditPage != 0) {
```

Comment:
you need to use <form:hidden> tag on jsp to store id of the image.
`<form:hidden path="product.image.id" />`